src/api/ws.ts

```diff
@@ -1,29 +1,19 @@
-import { IpcMain, WebContents } from 'electron';
+import { WebContents } from 'electron';
 import WebSocket from 'ws';
 
-import { JsonRPCResponse } from '@types';
+import { handleRequest } from './api';
 
-import { handleRequest, handleResponse } from './api';
-
-export const runAPI = (ipcMain: IpcMain, webContents: WebContents) => {
+export const runAPI = (webContents: WebContents) => {
   console.debug('Spinning up WS');
   const ws = new WebSocket.Server({ host: 'localhost', port: 8000 });
   ws.on('connection', (socket) => {
-    const reply = (response: JsonRPCResponse) =>
+    socket.on('message', async (data) => {
```

Comment:
Code smell: This looks a callback hell. Refactor plz.

Reply:
Not sure how we avoid this. I've reduced the callback nesting heavily with this refactor.